### DISCUSSION TITLE
3D mouse input support

### DISCUSSION THREAD

**Discussion by brenocq** [brenocq/implot3d#38] [General]
### This discussion was converted into issue #105

Add support for 3D mouse input to interactively manipulate 3D plots (e.g., rotate, pan, and zoom). This would improve usability and provide an intuitive way to explore 3D data.

![Screenshot_20241223-074706.png](https://github.com/user-attachments/assets/92f98a6b-b7e7-4689-bcb0-77a8b4384a61)

**Comment by dimateos** [comment]
This reminds me of horizontal scrolling when you hold SHIFT (e.g. in browsers).

Not sure what is usually the case:
* this 3D device issues horizontal scroll
* this 3D device emulate SHIFT+scroll
* regular mouse issues horizontal axis when the user is holding SHIFT
* the browsers interprets SHIFT+scroll on regular mouse as horizontal
> for the third axis idk SHIFT+ALT or SHIFT+CTRL?

My point is that you can probably code this already for a regular mouser + modifier(s) key.
But maybe modifier keys are better suited for other tasks tho

**Reply by brenocq** [reply]
That's a very interesting idea! Maybe we could investigate how other 3d plot libraries use mod keys to get some ideas of mod keys for implot3d.

If we decide to implement #17 we would indeed need to implement some mod keys to move in 3D (like blender-style, solidworks-style, etc)

**Comment by brenocq** [comment]
Hi @dimateos, I'm converting all feature discussions to issues since GitHub has better support for task tracking (and copilot agent execution) with issues. This discussion was converted into #105
